src/components/Users.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+   <table class="table">
+    <thead>
+      <tr>
+        <th>Title</th>
+        <th>
+            <td>
+              <div>User Name</div>
+            </td>
+            <td> 
+              <div class="btn-holder"><el-button type="primary" @click="sortByName">Sort by User Name</el-button></div>
+            </td>
+        </th>
+        <th>User selected</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr  
+        v-for="user in sortedUsers"
+        :key="user.id"
+      >
+        <td>{{user.name}}</td>
+        <td>{{user.username}}</td>
+        <td>          
+          <label 
+          :for="'checkbox_' + user.id"
+          >
+            <input 
+              type="checkbox" 
+              class="checkbox"
+              :name="'checkbox_' + user.id"
+              v-model="checkedUser[user.id]"
+              @change="checkedUserId(user.id)">
+            <span  v-if="checkedUser[user.id]">Primary</span>            
+          </label>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+
+<script>
+export default {
+  /**
+   * Name.
+   */
+  name: "Users",
+
+  /**
+   * Props.
+   */
+  props: {
+    /**
+     * Users
+     */
+    users: {
+      type: Array, 
+      required: true
+    },
+  },
+
+  /**
+   * Reactive properties.
+   * @returns {{}}
+   */
+  data() {
+    return {
+      checkedUser: {},
+      sortUsersByName: [],
+    };
+  },
+
+  /**
+   * Computed.
+   */
+  computed: {
+
+    /**
+    * Sorted users.
+    */
+    sortedUsers() {
+      return this.sortUsersByName.length ? this.sortUsersByName : this.users;
```

Comment:
save this.users into current state this.currentUsers as soon as we receive props.
Modify this.currentUsers whenever you click sort.